tips/TIP-0020/tip-0020.md

```diff
@@ -0,0 +1,497 @@
+        <summary>Ed25519 Signature</summary>
+        <table>
+          <tr>
+            <th>Name</th>
+            <th>Type</th>
+            <th>Description</th>
+          </tr>
+          <tr>
+            <td>Signature Type</td>
+            <td>uint8</td>
+            <td>
+              Set to <strong>value 0</strong> to denote an <i>Ed25519 Signature</i>.
+            </td>
+          </tr>
+          <tr>
+            <td>Public key</td>
+            <td>ByteArray[32]</td>
+            <td>The Ed25519 public key of the signature.</td>
+          </tr>
+          <tr>
+            <td>Signature</td>
+            <td>ByteArray[64]</td>
+            <td>The Ed25519 signature signing the Blake2b-256 hash of the serialized <i>Transaction Essence</i>.</td>
+          </tr>
+        </table>
+      </details>
+    </td>
+  </tr>
+</table>
+
+##### Unlock syntactic validation
+
+* `Signature` must contain an _Ed25519 Signature_.
+* The _Signature Unlock_ must be unique, i.e. there must not be any other _Signature Unlocks_ in the `Unlocks` field of the transaction payload with the same signature.
+
+#### Reference Unlock
+
+The <i>Reference Unlock</i> defines an <i>Unlock</i> which references a previous <i>Unlock</i> (which must not be another <i>Reference Unlock</i>). It **must** be used if multiple inputs can be unlocked via the same <i>Unlock</i>. It is serialized as follows:
+
+<table>
+  <tr>
+    <th>Name</th>
+    <th>Type</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>Unlock Type</td>
+    <td>uint8</td>
+    <td>
+      Set to <strong>value 1</strong> to denote a <i>Reference Unlock</i>.
+    </td>
+  </tr>
+  <tr>
+    <td>Reference</td>
+    <td>uint16</td>
+    <td>Represents the index of a previous unlock.</td>
+  </tr>
+</table>
+
+##### Unlock syntactic validation
+
+* The _Reference Unlock_ at index i must have `Reference` < i and the unlock at index `Reference` must be a _Signature Unlock_.
+
+Example:
+Consider a <i>Transaction Essence</i> containing the <i>UTXO Inputs</i> 0, 1 and 2, where 0 and 2 are both spending outputs belonging to the same Ed25519 address A and 1 is spending from a different address B. This results in the following structure of the <i>Unlocks</i> part:
+| Index | Unlock                                                                             |
+| ----- | ---------------------------------------------------------------------------------------- |
+| 0     | A _Signature Unlock_ holding the Ed25519 signature for address A.                  |
+| 1     | A _Signature Unlock_ holding the Ed25519 signature for address B.                  |
+| 2     | A _Reference Unlock_ which references 0, as both require the same signature for A. |
+
+## Validation
+
+A <i>Transaction Payload</i> has different validation stages, since some validation steps can only be executed when certain information has (or has not) been received. We therefore distinguish between syntactic and semantic validation
```

Comment:
```suggestion
A <i>Transaction Payload</i> has different validation stages, since some validation steps can only be executed when certain information has (or has not) been received. We therefore distinguish between syntactic and semantic validation.
```

Reply:
Added in https://github.com/iotaledger/tips/pull/40/commits/d9370b586a61719331d7fd1fccea365ee5c5d137